p2p/ping.go

```diff
@@ -83,14 +83,22 @@ func pingPeer(ctx context.Context, svc *ping.PingService, p peer.ID, callback fu
 func pingPeerOnce(ctx context.Context, svc *ping.PingService, p peer.ID,
 	logFunc func(context.Context, ping.Result), callback func(peer.ID, host.Host),
 ) {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	ticker := time.NewTicker(time.Second * 30)
+	defer ticker.Stop()
 
+	pingChan, closeStream := getPingChan(ctx, svc, p)
+	defer closeStream()
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case result := <-svc.Ping(ctx, p): // Only ping once to always use "best" connection.
```

Comment:
This creates new streams for each leaving existing streams open.

Reply:
why then not just close the existing streams. Simplest sounds like new stream for each ping, making sure it is closed.

Reply:
Sounds good! Pushed the changes.